miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt

```diff
@@ -48,6 +57,24 @@ internal class MiniAppDownloader(
         miniAppStatus.saveDownloadedMiniApp(miniAppInfo)
     }
 
+    private suspend fun retrieveDownloadedVersionPath(miniAppInfo: MiniAppInfo): String? {
+        val versionPath = storage.getMiniAppVersionPath(miniAppInfo.id, miniAppInfo.version.versionId)
+
+        if (miniAppStatus.isVersionDownloaded(miniAppInfo.id, miniAppInfo.version.versionId, versionPath)) {
+            return if (verifier.verify(miniAppInfo.id, File(versionPath))) {
+                versionPath
+            } else {
+                Log.e(TAG, "Failed to verify the hash of the cached files. " +
+                        "The files will be deleted and the Mini App re-downloaded.")
+                storage.removeOutdatedVersionApp(miniAppInfo.id, NON_EXISTENT_VERSION_ID)
```

Comment:
Why do we use `NON_EXISTENT_VERSION_ID`? Should it be `miniAppInfo.version.versionId`?

Reply:
I understand it now. I forget my own implementation on `removeOutdatedVersionApp` 😆. Please ignore my previous comment.

I think it's better to remove the whole package (miniapp id directory) of this miniapp.

Reply:
You're right, it's kind of confusing that I'm using `removeOutdatedVersionApp` to remove the whole app... I've added a new function to remove the entire app folder: `removeApp`.